### ISSUE TITLE
Unable to handle more request

### ISSUE THREAD

**Issue by rabi07** [hyperxpro/SourceEngineQueryCacher#40]
sir ,Im using CS2 , only port udp 25017 open

with below config
Threads=100
StatsPPS=true
StatsbPS=true
GameUpdateInterval=1000
GameUpdateSocketTimeout=1000
ChallengeCodeTTL=5000
LocalServerIPAddress=0.0.0.0
LocalServerPort=9110
GameServerIPAddress=127.0.0.1
GameServerPort=27015
ReceiveBufferSize=65535
SendBufferSize=65535
UdpGro=true
A2sRule=true

The error shown below appears when a DDoS attack occurs.

------------ Dropping query request because Cache is not ready -----------
A2S_INFO: PooledUnsafeDirectByteBuf(ridx: 0, widx: 144, cap: 256)
A2S_PLAYER: PooledUnsafeDirectByteBuf(ridx: 0, widx: 0, cap: 256)
-----------------------

im getting p/s more then 30k and b/s around 50k 
any way to fix it?
Thanks you

![image](https://github.com/hyperxpro/SourceEngineQueryCacher/assets/38426299/0e121f84-1b0d-4f38-8395-57405c4e979a)

**Comment by hyperxpro** [comment]
I have some optimizations in my plans. I'd need a server to test. Wanna help around?

**Comment by rabi07** [comment]
@hyperxpro  sure sir ,  ping me here savage5555 on discord

ill provide server

**Comment by Gawnz1** [comment]
@hyperxpro , Better and for maximum performance will be XDP, so if you have time and desire, rewrite it in XDP.
Rust + XDP can also be an option, I see that you have some repository (game oriented) already.

**Comment by hyperxpro** [comment]
@Gawnz1 Indeed, I do plan for XDP but `redbpf` is no longer being maintained. I have to use a new Rust library to make things working again.

**Comment by Gawnz1** [comment]
That's nice to hear! I usually use [LibXDP](https://github.com/xdp-project/xdp-tools/tree/master/lib/libxdp) / [LibBPF](https://github.com/xdp-project/xdp-tools/tree/master/lib) with C, since I don't know rust. You might take a look, if it fits for your project/language (If there is nothing for rust yet).
